### PR TITLE
KH & KB fixes

### PR DESCRIPTION
**Pull request by charisprose**

### CODE REVIEW

**Comment by jacobbeasley** [comment]
@charisprose  We should figure out why your PRs always seem to have 50+ commits in them. Maybe need to re-check out the repo or something.

File: dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html [hunk 1]
```diff
@@ -40,7 +40,7 @@ <h3 class="error" *ngIf="statusInvalid">{{benchmarkStatus}}</h3>
           </mat-card>
         </div>
 
-        <div class="col-xs-12 col-sm-12 col-md-6 col-lg-6 margin-top-10" style="margin: 0; padding-left:0; padding-right:0; height:150px; display: grid;!important; overflow: auto">
+        <div class="col-xs-12 col-sm-12 col-md-6 col-lg-5 margin-top-10" style="margin: 0; padding-left:0; padding-right:0; height:150px; display: grid;!important; ">
```
Comment (jacobbeasley) [inline dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html:43]: Why are we removing overflow: auto?

File: dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html [hunk 2]
```diff
@@ -122,7 +122,7 @@ <h3 class="error" *ngIf="statusInvalid">{{benchmarkStatus}}</h3>
                     </ng-container>
 
                     <mat-header-row *matHeaderRowDef="displayedColumns"></mat-header-row>
-                    <mat-row class="bench-audit-row" *matRowDef="let row; columns: displayedColumns" (click)="toReportDetails(row.id)"></mat-row>
+                    <mat-row class="bench-audit-row" *matRowDef="let row; columns: displayedColumns" (click)="toReportDetails(row.id)"  [ngStyle]="{'cursor': 'pointer'}"></mat-row>
```
Comment (jacobbeasley) [inline dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html:125]: I wonder if we should create a global "clickable" class or something for this kind of thing? @becky-intelletive  @kyle-intelletive @charisprose

File: dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter-report-details/kube-hunter-report-details.component.html [hunk 3]
```diff
@@ -19,6 +19,7 @@
               </div>
             </mat-card-header>
             <mat-card-content>
+              <div class="page-table">
```
Comment (jacobbeasley) [inline dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter-report-details/kube-hunter-report-details.component.html:22]: Why is this needed?

File: dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter.component.ts [hunk 4]
```diff
@@ -71,6 +72,11 @@ export class KubeHunterComponent implements OnInit, OnDestroy {
           this.allReportsForCluster = res.list;
           if (res.list[0] != null){
             this.daysPassed = Math.floor((Date.now() - res.list[0].createdAt) / (1000 * 60 * 60 * 24));
+            if (this.daysPassed >= 90){
+              this.ourAdvice = 'It has been ' + this.daysPassed + ' days since you ran Kube Hunter. You should run it again soon.';
+            } else {
+              this.ourAdvice = 'It has been ' + this.daysPassed + ' days since you last ran Kube Hunter. The current report is still valid.';
```
Comment (jacobbeasley) [inline dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter.component.ts:78]: Lets remove the "The current report is still valid" part. We don't know how frequently their security team wants them to run it.

File: dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter.component.html [hunk 5]
```diff
@@ -3,76 +3,70 @@
...
+          <div class="col-xs-12 col-sm-12 col-md-6 col-lg-5 margin-top-10" style=" margin: 0; padding-left:0; padding-right:0; height:150px; display: grid;!important;">
+            <mat-card appearance="outlined">
+              <mat-card-header>
+                <div class="page-card-title">Our Advice:</div>
+              </mat-card-header>
+              <mat-card-content>
+                <div class="page-card-header">
+                  <div class=" col-sm-8 no-padding">
+                    <p>
+                      {{ourAdvice}}
```
Comment (jacobbeasley) [inline dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter.component.html:52]: nice

File: dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter.component.ts (hunk 4 above)
Comment (jacobbeasley) [reply dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter.component.ts:78]: Oh, I see that's how it used to work, but yeah lets remove "The current report is still valid." anyways while we're working on this, thanks!

File: dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html (hunk 1 above)
Comment (charisprose) [reply dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html:43]: There was a vertical scroll appeared next to the card when shrinking to a smaller screen size (when card was small enough the title got moved to the next line).

File: dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter-report-details/kube-hunter-report-details.component.html (hunk 3 above)
Comment (charisprose) [reply dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter-report-details/kube-hunter-report-details.component.html:22]: This class was created when I was trying different things to make the table to show correctly in a smaller screen. I forgot to clean up. Since there's a overflow in this class, I can make it a inline style.

Comment (becky-intelletive) [reply dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter-report-details/kube-hunter-report-details.component.html:22]: There are classes for overflowing cards.

`<mat-card class=".scrollable-table-card">`
`  <div class=table-wrapper>`
`    [mat-table element goes here]`
`  </div>`
`  <mat-paginator></mat-paginator>` (if applicable)
`</mat-card>`

File: dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html [hunk 6]
```diff
@@ -6,7 +6,7 @@
     <div class="page-card">
       <div class="row">
 
-        <div class="col-xs-12 col-sm-12 col-md-6 col-lg-3 margin-top-10" style="margin: 0; padding-left:0; height:150px; display: grid;!important; overflow: auto">
```
Comment (becky-intelletive) [inline dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html:9]: When resized to small screens, are these tables scrollable?
I'd guess we'll have to add those classes (mentioned in a different comment)

File: dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html (hunk 1 above)
Comment (jacobbeasley) [reply dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html:43]: Without overflow:auto how does it look now on small screens?

Comment (jacobbeasley) [reply dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html:43]: I wonder if maybe it should allow horizontal scrolling but not vertical scrolling on small screens

File: dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html (hunk 2 above)
Comment (kyle-intelletive) [reply dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html:125]: We have a global CSS class called 'pointer' that applies the 'cursor: pointer' rule.

We should also not use inline styles from a best practices & maintainability standpoint

**Comment by becky-intelletive** [comment]
> @charisprose We should figure out why your PRs always seem to have 50+ commits in them. Maybe need to re-check out the repo or something.

I always re-sync my fork when the main repo has new changes. Maybe you could try that?

1. In GitHub, resync the project. Click the "sync fork" dropdown and update it to the main repo.
image

2. In your local repo, make sure your changes are committed or stashed.
3. Checkout your main branch: `git checkout main`
4. Retrieve the updated origin: `git fetch --all`
image

5. Reset your local main to match your upstream main: `git reset --hard origin/main`
image
**NOTE: this will remove ALL local commits, so make sure they're in a separate branch or you'll have to redo them**

6. Branch off of your updated main for future work. Never push to main (or complete PRs to main).

File: dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html (hunk 6 above)
Comment (charisprose) [reply dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html:9]: @becky-intelletive In the mobile size, info from these cards are visible (scrolls are not necessary)
Screenshot 2023-07-24 at 12 04 18 PM
As for adding classes, are you referring to adding class="col-xs-12 col-sm-12 col-md-6 col-lg-3 margin-top-10" as a global class?

File: dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter-report-details/kube-hunter-report-details.component.html (hunk 3 above)
Comment (becky-intelletive) [reply dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter-report-details/kube-hunter-report-details.component.html:22]: This isn't quite right yet. See my comment on where the classes should generally go.
It also doesn't scroll properly on a small screen yet.

File: dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html (hunk 6 above)
Comment (becky-intelletive) [reply dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html:9]: (Same goes for the Kube Hunter page)

File: dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter.component.ts (hunk 4 above)
Comment (becky-intelletive) [reply dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter.component.ts:78]: Bump

**Review by becky-intelletive — CHANGES_REQUESTED** [review]

File: dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter-report-details/kube-hunter-report-details.component.html (hunk 3 above)
Comment (charisprose) [reply dash/frontend/src/app/modules/private/pages/cluster/kube-hunter/kube-hunter-report-details/kube-hunter-report-details.component.html:22]: This is how the scroll looks like in a mobile screen.
Screenshot 2023-07-24 at 2 16 54 PM

File: dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html (hunk 6 above)
Comment (charisprose) [reply dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html:9]: If you're referring to the width of the 3 cards, Test status and vulnerabilities cards have been more narrow that way to accommodate Our Advice card. I can adjust Our Advice card to fix the cutoff text.

Comment (becky-intelletive) [reply dash/frontend/src/app/modules/private/pages/cluster/kube-bench/kube-bench.component.html:9]: That should be handled by the column sizing, which should be uniform for a small screen (col-xs-12 for all elements).

**Comment by charisprose** [comment]
Just gave another push. Should be good now.

Charis Prose
Software Engineer of Intelletive
***@***.***

**Review by becky-intelletive — APPROVED** [review]
